cdap-tms/src/main/java/co/cask/cdap/messaging/store/hbase/HBaseTableFactory.java

```diff
@@ -136,6 +140,96 @@ public PayloadTable createPayloadTable(String tableName) throws IOException {
     );
   }
 
+  @Override
+  public void upgradeMessageTable(String tableName) throws IOException {
+    try {
+      upgradeCoProcessor(tableUtil.createHTableId(NamespaceId.SYSTEM, tableName),
+                         tableUtil.getMessageTableRegionObserverClassForVersion());
+    } catch (TableNotFoundException ex) {
```

Comment:
upgradeCoProcessor calls disableTable (which disables the table and also handles the case where the table is disabled already) before updating the coprocessor.